src/http_client_async.zig

```diff
@@ -2112,12 +2139,15 @@ pub fn onData(this: *HTTPClient, comptime is_ssl: bool, incoming_data: []const u
 }
 
 pub fn closeAndAbort(this: *HTTPClient, reason: JSC.JSValue, comptime is_ssl: bool, socket: NewHTTPContext(is_ssl).HTTPSocket) void {
-    socket.ext(**anyopaque).?.* = bun.cast(
-        **anyopaque,
-        NewHTTPContext(is_ssl).ActiveSocket.init(&dead_socket).ptr(),
-    );
-    socket.close(0, null);
+    log("closeAndAbort", .{});
     this.fail(error.Aborted, reason);
+    if (!socket.isClosed()) {
```

Comment:
you sure there's no weird memory stuff due to the ordering of this?

Reply:
This is the original order of this.closeAndFail, now I changed to avoid any issues, I think this got swift places when I was refactoring last time.